controller/space_iterations_test.go

```diff
@@ -244,163 +245,141 @@ func (rest *TestSpaceIterationREST) TestFailListIterationsByMissingSpace() {
+			case 0, 1, 2:
+				wi.Fields[workitem.SystemState] = workitem.SystemStateNew
+				wi.Fields[workitem.SystemIteration] = fxt.Iterations[1].ID.String()
+				wi.Type = workitem.SystemFeature
+			case 3, 4, 5:
+				wi.Fields[workitem.SystemState] = workitem.SystemStateNew
+				wi.Fields[workitem.SystemIteration] = fxt.Iterations[1].ID.String()
+				wi.Type = workitem.SystemTask
+			case 6, 7:
+				wi.Fields[workitem.SystemState] = workitem.SystemStateClosed
+				wi.Fields[workitem.SystemIteration] = fxt.Iterations[1].ID.String()
+				wi.Type = workitem.SystemFeature
+			case 8, 9:
+				wi.Fields[workitem.SystemState] = workitem.SystemStateClosed
+				wi.Fields[workitem.SystemIteration] = fxt.Iterations[1].ID.String()
+				wi.Type = workitem.SystemTask
+			case 10, 11, 12, 13:
+				wi.Fields[workitem.SystemState] = workitem.SystemStateNew
+				wi.Fields[workitem.SystemIteration] = fxt.Iterations[3].ID.String()
+				wi.Type = workitem.SystemFeature
+			case 14, 15, 16, 17:
+				wi.Fields[workitem.SystemState] = workitem.SystemStateNew
+				wi.Fields[workitem.SystemIteration] = fxt.Iterations[3].ID.String()
+				wi.Type = workitem.SystemTask
+			case 18, 19, 20, 21, 22:
+				wi.Fields[workitem.SystemState] = workitem.SystemStateClosed
+				wi.Fields[workitem.SystemIteration] = fxt.Iterations[4].ID.String()
+				wi.Type = workitem.SystemFeature
+			case 23, 24, 25, 26, 27:
+				wi.Fields[workitem.SystemState] = workitem.SystemStateClosed
+				wi.Fields[workitem.SystemIteration] = fxt.Iterations[4].ID.String()
+				wi.Type = workitem.SystemTask
+			}
+			return nil
+		}),
+	)
 
 	svc, ctrl := rest.UnSecuredController()
 	// when
-	_, cs := test.ListSpaceIterationsOK(rest.T(), svc.Context, svc, ctrl, spaceInstance.ID, nil, nil)
+	_, cs := test.ListSpaceIterationsOK(rest.T(), svc.Context, svc, ctrl, fxt.Spaces[0].ID, nil, nil)
 	// then
-	require.Len(rest.T(), cs.Data, 4)
+	require.Len(rest.T(), cs.Data, 5)
 	for _, iterationItem := range cs.Data {
-		if uuid.Equal(*iterationItem.ID, iteration1.ID) {
+		if uuid.Equal(*iterationItem.ID, fxt.Iterations[1].ID) {
 			assert.Equal(rest.T(), 5, iterationItem.Relationships.Workitems.Meta[KeyTotalWorkItems])
 			assert.Equal(rest.T(), 2, iterationItem.Relationships.Workitems.Meta[KeyClosedWorkItems])
-		} else if uuid.Equal(*iterationItem.ID, iteration2.ID) {
+		} else if uuid.Equal(*iterationItem.ID, fxt.Iterations[2].ID) {
 			// we expect these counts should include that of child iterations too.
 			expectedTotal := 0 + 4 + 5  // sum of all items of self + child + grand-child
 			expectedClosed := 0 + 0 + 5 // sum of closed items self + child + grand-child
 			assert.Equal(rest.T(), expectedTotal, iterationItem.Relationships.Workitems.Meta[KeyTotalWorkItems])
 			assert.Equal(rest.T(), expectedClosed, iterationItem.Relationships.Workitems.Meta[KeyClosedWorkItems])
-		} else if uuid.Equal(*iterationItem.ID, childOfIteration2.ID) {
+		} else if uuid.Equal(*iterationItem.ID, fxt.Iterations[2].ID) {
```

Comment:
@baijum this if statement is same as the one on line number 316.

Reply:
Goo catch! Fixed. 0db6d6763efc22096da8a463a353388b235c3e5d